AppCenter/AppCenter/Internals/Channel/MSChannelUnitDefault.m

```diff
@@ -492,20 +492,24 @@ - (void)setEnabled:(BOOL)isEnabled andDeleteDataOnDisabled:(BOOL)deleteData {
   });
 }
 
-- (void)suspend {
-  if (!self.suspended) {
-    MSLogDebug([MSAppCenter logTag], @"Suspend channel for group Id %@.",
-               self.configuration.groupId);
-    self.suspended = YES;
+- (void)pauseWithToken:(id <NSObject>)token {
+  [self.pausedTokens addObject:token];
+  MSLogDebug([MSAppCenter logTag], @"Pause token %@ added to channel with group Id %@.",
+             token, self.configuration.groupId);
+  if (!self.paused) {
+    MSLogDebug([MSAppCenter logTag], @"Pause channel for group Id %@.", self.configuration.groupId);
+    self.paused = YES;
     [self resetTimer];
   }
 }
 
-- (void)resume {
-  if (self.suspended && self.enabled) {
-    MSLogDebug([MSAppCenter logTag], @"Resume channel for group Id %@.",
-               self.configuration.groupId);
-    self.suspended = NO;
+- (void)resumeWithToken:(id <NSObject>)token {
+  [self.pausedTokens removeObject:token];
+  MSLogDebug([MSAppCenter logTag], @"Pause token %@ removed from channel with group Id %@.",
+             token, self.configuration.groupId);
+  if ([self.pausedTokens count] == 0) {
```

Comment:
It only makes sense if tokens are services like channel, storage and ingestion. What if it is a target token? Not supposed to be used for transmission target pause/resume?